Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1674,6 +1678,116 @@ JS::GCPtr<CSSSupportsRule> Parser::convert_to_supports_rule(Rule& rule)
+        return {};
+    }
+
+    if (!rule.block()) {
+        dbgln_if(CSS_PARSER_DEBUG, "Failed to parse @property rule: No block.");
+        return {};
+    }
+
+    auto prelude_stream = TokenStream { rule.prelude() };
+    prelude_stream.discard_whitespace();
+    auto& token = prelude_stream.consume_a_token();
+    if (!token.is_token()) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property has invalid prelude, prelude = {}; discarding.", rule.prelude());
+        return {};
+    }
+
+    auto name_token = token.token();
+    prelude_stream.discard_whitespace();
+
+    if (prelude_stream.has_next_token()) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property has invalid prelude, prelude = {}; discarding.", rule.prelude());
+        return {};
+    }
+
+    if (!name_token.is(Token::Type::Ident)) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property name is invalid: {}; discarding.", name_token.to_debug_string());
+        return {};
+    }
+
+    if (!name_token.ident().starts_with_bytes("--"sv)) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property name doesn't start with '--': {}; discarding.", name_token.ident());
+        return {};
+    }
+
+    auto const& name = name_token.ident();
+
+    auto block_tokens = rule.block()->values();
+    auto block_stream = TokenStream { block_tokens };
+
+    auto list_of_declarations = parse_a_list_of_declarations(block_stream);
+
+    auto syntax_maybe = Optional<FlyString> {};
+    auto inherit_maybe = Optional<bool> {};
+    auto initial_value_maybe = Optional<FlyString> {};
+
+    for (auto const& declOrRule : list_of_declarations) {
+        if (!declOrRule.is_declaration()) {
+            dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Only declarations allowed in @property block.; discarding.");
+            return {};
+        }
+        auto const& decl = declOrRule.declaration();
+
+        if (decl.name() == "syntax") {
+            if (syntax_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"syntax\" in @property block; discarding.");
+                return {};
+            }
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::String)) {
+                syntax_maybe = decl.values()[0].token().string();
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"syntax\": {}; discarding.", decl.to_string());
+            }
+        }
+        if (decl.name() == "inherits") {
+            if (inherit_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"inherits\" in @property block; discarding.");
+                return {};
+            }
+            // Feels hacky using Ident here, but this is what the tokenizer gives us.
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::Ident)) {
```

Comment:
Same point about the token count, but also, you might want to know about `ComponentValue::is_ident("hello"sv)`.